### PR TITLE
sih: move to PX4Accelerometer/PX4Gyroscope/PX4Magnetometer/PX4Barometer helpers

### PR DESCRIPTION
**Pull request by dagar**
@romain-chiap

The only real change to check (and test) here is `void Sih::send_IMU()` that's been updated to use helper classes for publishing raw sensor data. Using these are important so that the same filtering and integration is applied to the sensor data.

### CODE REVIEW

**Comment by dagar** [comment]
Discovered that astyle was chocking on PRINT_MODULE_DESCRIPTION, which is how this slipped through CI.

File: src/modules/sih/sih.hpp [hunk 1]
```diff
@@ -98,18 +96,12 @@ class Sih : public ModuleBase<Sih>, public ModuleParams
...
-	struct sensor_mag_s             _sensor_mag {};
-	orb_advert_t                    _sensor_mag_pub{nullptr};
-	// to publish the sensor gyroscope
-	struct sensor_gyro_s            _sensor_gyro {};
-	orb_advert_t                    _sensor_gyro_pub{nullptr};
-	// to publish the sensor accelerometer
-	struct sensor_accel_s           _sensor_accel {};
-	orb_advert_t                    _sensor_accel_pub{nullptr};
+	// simulated sensor instances
+	PX4Accelerometer _px4_accel{ 1311244, ORB_PRIO_DEFAULT, ROTATION_NONE }; // 1311244: DRV_ACC_DEVTYPE_ACCELSIM, BUS: 1, ADDR: 1, TYPE: SIMULATION
```
Comment (bkueng) [inline src/modules/sih/sih.hpp]: Can you use a macro/method for the ID?

Comment (dagar) [reply src/modules/sih/sih.hpp]: We could build them up piece by piece with multiple calls setting each thing, but it didn't seem worth it given we ultimately still have magic numbers floating around.

https://github.com/PX4/Firmware/blob/d2c824c5346f27907043efae6b81e3bf73ec51c8/ROMFS/px4fmu_common/init.d-posix/rcS#L99

Comment (bkueng) [reply src/modules/sih/sih.hpp]: > We could build them up piece by piece with multiple calls setting each thing

I still prefer that.

Comment (dagar) [reply src/modules/sih/sih.hpp]: I'll come back to it in another pass system wide.

**Pull request merged**